modules/ROOT/pages/logging-in-mule.adoc

```diff
@@ -0,0 +1,284 @@
+
+The default exception handler is shown below:
+
+----
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.launcher.api.log4j2;
+
+import com.lmax.disruptor.ExceptionHandler;
+
+import org.apache.logging.log4j.status.StatusLogger;
+
+/**
+ * Implementation of {@link com.lmax.disruptor.ExceptionHandler} to be used when async loggers fail to log their messages. It will
+ * log this event using the {@link org.apache.logging.log4j.status.StatusLogger}
+ *
+ * @since 3.6.0
+ */
+public class AsyncLoggerExceptionHandler implements ExceptionHandler {
+
+  private static final StatusLogger LOGGER = StatusLogger.getLogger();
+
+  @Override
+  public void handleEventException(Throwable ex, long sequence, Object event) {
+    LOGGER.error("Failed to asynchronously log message: " + event, ex);
+  }
+
+  @Override
+  public void handleOnStartException(Throwable ex) {
+    LOGGER.error("Failed to start asynchronous logger", ex);
+  }
+
+  @Override
+  public void handleOnShutdownException(Throwable ex) {
+    LOGGER.error("Failed to stop asynchronous logger", ex);
+  }
+}
+----
+
+There is a performance-reliability trade-off between asynchronous and synchronous logging. If the risk of losing log messages is a serious issue, then configure your loggers to be synchronous. You can have a mix of both synchronous and asynchronous logging.
+
+== Configuration Reloading
+
+Mule runtime uses Log4j 2 as the framework for logging. To add or change the monitoring interval at which Log4j 2 checks for changes, add the `monitorInterval` to the `log4j2.xml` configuration file:
+
+. In the Mule installation folder, open `conf/log4j2.xml`.
+. Add `<Configuration monitorInterval="15">` after the first line. For example:
++
+[source,xml,linenums]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration monitorInterval="15">
+    <Appenders>
+       <Console name="Console" target="SYSTEM_OUT">
+           <PatternLayout pattern="%-5p %d [%t] %c: %m%n"/>
+       </Console>
+    </Appenders>
+...
+----
++
+In this example, Log4j 2 checks for changes every 15 seconds.
++
+. Save the file, and restart the Mule server.
+
+For more information, see the Apache documentation on automatically configuring Log4j 2.
+
+== Debugging Projects that Use the HTTP Connector
+
+To debug projects that use the HTTP connector, you might want to make the logging more verbose than usual and track all of the behavior of both the Http-Listener and Http-Request connectors in your project. See https://docs.mulesoft.com/connectors/http/http-troubleshooting for information.
```

Comment:
Should we add xref link here or it's not possible?

Reply:
OK